spec/controllers/auth_spec.rb

```diff
@@ -0,0 +1,34 @@
+require "rails_helper"
+
+describe AuthController, type: :controller do
+  describe "POST # register" do
+    context "with valid params" do
+      let (:params) do
+        {
+          email: "tuannguyensn2001a@gmail.com",
+          username: "tuannguyensn2001",
+          password: "tuannguyensn2001"
+        }
+      end
+      it "return response success" do
+        post :register, params: params
+        expect(response).to have_http_status(:ok)
+        expect(User.count).to eq(1)
+        expect(User.first.email).to eq("tuannguyensn2001a@gmail.com")
```

Comment:
This controller is being tested with the Auth::Register logic when it shouldn't. You have to stub the service

Reply:
you mean that service expect to be called ?

Reply:
yup, stub the service to be called with the correct params and mock the returning data